app_dart/lib/src/request_handlers/refresh_github_commits.dart

```diff
@@ -59,22 +59,16 @@ class RefreshGithubCommits extends ApiRequestHandler<Body> {
     final DatastoreService datastore = datastoreProvider(config.db);
 
     for (String branch in await config.flutterBranches) {
-      // TODO(keyonghan): use time of latest commit for [hours] in github API query, https://github.com/flutter/flutter/issues/54284
-      int hours = config.existingBranchHours;
-
-      if (branch != 'master') {
-        final Set<Commit> commits = await datastore
-            .queryRecentCommits(limit: 1, branch: branch)
-            .toSet();
-
-        /// Set [hours] to be longer if a new branch is found.
-        if (commits.isEmpty) {
-          hours = config.newBranchHours;
-        }
+      final List<Commit> lastCommit =
+          await datastore.queryRecentCommits(limit: 1, branch: branch).toList();
+      int lastCommitTimestampMills = 0;
+      if (lastCommit.isNotEmpty) {
+        lastCommitTimestampMills = lastCommit[0].timestamp;
       }
 
-      final List<RepositoryCommit> commits =
-          await githubService.listCommits(slug, branch, hours, config);
+      final List<RepositoryCommit> commits = await githubService.listCommits(
```

Comment:
What happens if last commit is empty? should we just return here, most likely and empty commit means nothing to process.

Reply:
If `lastProcessedCommit ` is empty, it is possible that a new release branch is available. We still need to query github to insert the corresponding data.

Reply:
If this is the only case I'd recommend adding docs to make it clear lastProcessedCommit==0 means new branch.

Reply:
Yes. That is the only case. I have added the doc.